### ISSUE TITLE
Modification check failed for some materials after a GoCd restart

### ISSUE THREAD

**Issue by aamattos** [gocd/gocd#8847]
##### Issue Type

- Bug Report

##### Summary

After a GoCd restart several materials show the error "Modification check failed for material"

##### Environment

Alpine Docker based GoCd server.

###### Basic environment details

* Go Version: `20.9.0 (12335-cccb4960bd…79277093295ed5a18573ed)`
* JAVA Version: `15`
* OS: `Linux 3.10.0-862.14.4.el7.x86_64`

##### Steps to Reproduce

After a restart, several materials show a modification check failed error, and related pipelines cannot be triggered

##### Expected Results
Trigger a pipeline should fetch the latest commit.

##### Actual Results

Materials show the error:

Error performing command: --- Command ---
git rev-parse --abbrev-ref HEAD
--- Environment ---
{}
--- INPUT ----

--- EXIT CODE (128) ---
--- STANDARD OUT ---
HEAD
--- STANDARD ERR ---
STDERR: fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
STDERR: Use '--' to separate paths from revisions, like this:
STDERR: 'git <command> [<revision>...] -- [<file>...]'
---
 

##### Any other info

We are using Github Enterprise, and materials with webhooks enabled. (no polling)
I checked the GoCd Server Log, and found no mention to those materials.

**Comment by maheshp** [comment]
Are you still facing this issue? Ideally, a restart of the server should not lead to material update failures. 

Can you Test Connection on the material configuration page?

**Comment by philnichol** [comment]
I'm getting the same, similarly after rebooting the server. Test connection fails, @aamattos did you ever manage to resolve this?

**Comment by philnichol** [comment]
Managed to resolve this by deleting the repo directory under /var/lib/go-server/pipelines/flyweight/<REPO_SID>
you can get the repo sid from the error message
Got the answer from this thead: https://groups.google.com/g/go-cd/c/_oPMKzWA1JU/m/06erfgw4X60J

TL;DR:
Aravind SA:
It's safe to move /var/lib/go-server/pipelines/flyweight/9a078662-bdb1-4862-8d54-56739da3e70c out of there, and the Go Server will check out the repo again once you've done that. Obviously, not knowing what happened, I can't tell you whether it will work. But, it's worth trying.
